### PR TITLE
Bento Carousel: `controls` prop

### PR DESCRIPTION
**Pull request by caroqliu**
The prop is critical for `amp-inline-gallery-thumbnails`:
`controls` = `"always"`, `"never"`, or `"auto"`
- `always`: Show arrows
- `never`: Hide arrows
- `auto`: Show arrows until touch interaction, then hide

Partial for #28284

### CODE REVIEW

File: extensions/amp-base-carousel/1.0/base-carousel.js [hunk 1]
```diff
@@ -80,13 +98,26 @@ function BaseCarouselWithRef(
...
+  }, [hadTouch, controls]);
+
   return (
     <ContainWrapper size={true} layout={true} paint={true} {...rest}>
       <Scroller
         loop={loop}
         restingIndex={currentSlide}
         setRestingIndex={setRestingIndex}
         ref={scrollRef}
+        onTouchStart={() => setHadTouch(true)}
```
Comment (dvoytenko) [inline extensions/amp-base-carousel/1.0/base-carousel.js:120]: So, is this mobile only feature? Or should we use `onPointerDown` instead?

File: extensions/amp-base-carousel/1.0/storybook/Basic.js [hunk 2]
```diff
@@ -30,8 +30,9 @@ export const _default = () => {
   const height = number('height', 225);
   const slideCount = number('slide count', 5, {min: 0, max: 99});
   const colorIncrement = Math.floor(255 / (slideCount + 1));
+  const controls = select('show controls', ['auto', 'always', 'never']);
```
Comment (dvoytenko) [inline extensions/amp-base-carousel/1.0/storybook/Basic.js:33]: Probably worth hoisting a const with this array up?

**Review by dvoytenko — APPROVED** [review]
LG, but a couple of q

File: extensions/amp-base-carousel/1.0/base-carousel.js (hunk 1 above)
Comment (caroqliu) [reply extensions/amp-base-carousel/1.0/base-carousel.js:120]: Mobile only - 0.1 looks for `touchstart` also

File: extensions/amp-base-carousel/1.0/storybook/Basic.js (hunk 2 above)
Comment (caroqliu) [reply extensions/amp-base-carousel/1.0/storybook/Basic.js:33]: Good point, done.

**Pull request merged**